### PR TITLE
Adjusted OpenMP flags

### PR DESCRIPTION
**Pull request by ravil-mobile**
* bumped min. required CMake version to 3.10
to guarantee CMake support of OpenMP
* removed custom FindOpenMP.cmake from SeisSol
* removed hard-coded openmp compiler and linker flags

Tested with **gcc**, **intel** and **clang** and it worked well for both c++ and fortran compilers

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -195,9 +199,19 @@ endif()
...
 if (OPENMP)
   find_package(OpenMP REQUIRED)
-  target_link_libraries(SeisSol-lib PUBLIC OpenMP::OpenMP_CXX)
-  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${OpenMP_Fortran_FLAGS}")
-  target_compile_definitions(SeisSol-lib PUBLIC OMP OMPI_SKIP_MPICXX)
+  target_link_libraries(SeisSol-gencode PRIVATE OpenMP::OpenMP_CXX)
+  target_link_libraries(SeisSol-lib PUBLIC OpenMP::OpenMP_Fortran)
+
+  find_package(OpenMP_Offloading QUIET)
+  # Note: there is a conflict between OpenMP offloading and libsxmm
```
Comment (krenzland) [inline CMakeLists.txt:206]: Maybe add a comment here why that is

File: CMakeLists.txt [hunk 2]
```diff
@@ -309,19 +323,29 @@ if (ADDRESS_SANITIZER_DEBUG)
   )
 endif()
 
+
+# Note: we need to provide a support for Eigen backend
+#       for both gencode and SeisSol
+target_include_directories(SeisSol-gencode PUBLIC
```
Comment (krenzland) [inline CMakeLists.txt:329]: Hm, why is this PUBLIC? Shouldn't this be private?

File: CMakeLists.txt [hunk 3]
```diff
@@ -309,19 +323,29 @@ if (ADDRESS_SANITIZER_DEBUG)
...
+    ${CMAKE_CURRENT_SOURCE_DIR}/submodules/eigen3
+    ${CMAKE_CURRENT_SOURCE_DIR}/submodules/yateto/include
+)
+
+target_include_directories(SeisSol-gencode PUBLIC
+    ${CMAKE_CURRENT_SOURCE_DIR}/submodules/yateto/include
```
Comment (krenzland) [inline CMakeLists.txt:335]: Why is this here twice?

File: cmake/FindOpenMP_Offloading.cmake [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+#  email: ravil.dorozhinskii@tum.de
+#
+
+include(FindPackageHandleStandardArgs)
+include(CheckCXXSourceRuns)
+include(CMakePushCheckState)
+
+cmake_push_check_state()
+
+set(_PARENT_CMAKE_CXX_STANDARD ${CMAKE_CXX_STANDARD})
```
Comment (krenzland) [inline cmake/FindOpenMP_Offloading.cmake:22]: Why is this needed? We are using CMAKE_CXX_STANDARD 17 anyway?

**Review by krenzland — CHANGES_REQUESTED** [review]
Cool! Looks mostly good, some comments

File: CMakeLists.txt (hunk 2 above)
Comment (ravil-mobile) [reply CMakeLists.txt:329]: So, as the comment says we need eigen (and yateto) headers for the code generated by `yateto` (in case of Eigen backend) and `SeisSol-lib`. If it is `PRIVATE` then the includes will be applied only for `SeisSol-gencode` target.

File: CMakeLists.txt (hunk 3 above)
Comment (ravil-mobile) [reply CMakeLists.txt:335]: you're right. I will remove it

File: cmake/FindOpenMP_Offloading.cmake (hunk 4 above)
Comment (ravil-mobile) [reply cmake/FindOpenMP_Offloading.cmake:22]: I guess I can remove `CMAKE_CXX_STANDARD` from this file. I started with a copy of `FindFILESYSTEM.cmake` file and forgot to remove it. 

In case of `FindFILESYSTEM.cmake`, _in general_, one doesn't know whether  `CMAKE_CXX_STANDARD 17` has already been set. It was just a generalization in case of somebody decides to re-use the file for other sub-projects.

File: CMakeLists.txt (hunk 1 above)
Comment (ravil-mobile) [reply CMakeLists.txt]: Added
```
  # Note: there may be a conflict between OpenMP offloading and libsxmm.
  # A collision can happen due to -march flags for CPUs and OpenMP GPU target
  # (e.g. with clang -Xopenmp-target -march=sm_75). Therefore, we separate
  # compilations for the source and generated codes and apply different
  # imported OpenMP targets to them.
```

File: CMakeLists.txt (hunk 2 above)
Comment (ravil-mobile) [reply CMakeLists.txt]: Yes, we are using. I mean the path to eigen will be applied to `SeisSol-gencode` and everything which depends on `SeisSol-gencode`. In our case, it is `SeisSol-lib` and so on (SeisSol-bin, SeisSol-proxy, tests, etc.).

**Review by krenzland — APPROVED** [review]
